src/main/scala/io/iohk/ethereum/utils/RLP.scala

```diff
@@ -0,0 +1,276 @@
+package io.iohk.ethereum.utils
+
+import java.nio.ByteBuffer
+
+import scala.annotation.{switch, tailrec}
+import scala.collection.immutable.Queue
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Recursive Length Prefix (RLP) encoding.
+  * <p>
+  * The purpose of RLP is to encode arbitrarily nested arrays of binary data, and
+  * RLP is the main encoding method used to serialize objects in Ethereum. The
+  * only purpose of RLP is to encode structure; encoding specific atomic data
+  * types (eg. strings, integers, floats) is left up to higher-order protocols; in
+  * Ethereum the standard is that integers are represented in big endian binary
+  * form. If one wishes to use RLP to encode a dictionary, the two suggested
+  * canonical forms are to either use &#91;&#91;k1,v1],[k2,v2]...] with keys in
+  * lexicographic order or to use the higher-level Patricia Tree encoding as
+  * Ethereum does.
+  * <p>
+  * The RLP encoding function takes in an item. An item is defined as follows:
+  * <p>
+  * - A string (ie. byte array) is an item - A list of items is an item
+  * <p>
+  * For example, an empty string is an item, as is the string containing the word
+  * "cat", a list containing any number of strings, as well as more complex data
+  * structures like ["cat",["puppy","cow"],"horse",[[]],"pig",[""],"sheep"]. Note
+  * that in the context of the rest of this article, "string" will be used as a
+  * synonym for "a certain number of bytes of binary data"; no special encodings
+  * are used and no knowledge about the content of the strings is implied.
+  * <p>
+  * See: https://github.com/ethereum/wiki/wiki/%5BEnglish%5D-RLP
+  *
+  * @author Roman Mandeleil
+  * @since 01.04.2014
+  */
+
+object RLP {
+  /**
+    * Reason for threshold according to Vitalik Buterin:
+    * - 56 bytes maximizes the benefit of both options
+    * - if we went with 60 then we would have only had 4 slots for long strings
+    * so RLP would not have been able to store objects above 4gb
+    * - if we went with 48 then RLP would be fine for 2^128 space, but that's way too much
+    * - so 56 and 2^64 space seems like the right place to put the cutoff
+    * - also, that's where Bitcoin's varint does the cutof
+    */
+  val SIZE_THRESHOLD: Int = 56
```

Comment:
Do you think we could use the naming conventions as proposed here: http://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods. That it upper camel case?